circle.yml

```diff
@@ -1,19 +1,44 @@
 machine:
-   environment:
-     ANDROID_HOME: /usr/local/android-sdk-linux
-     GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
-   java:
-     version: oraclejdk8
+  java:
+    version: oraclejdk8
+  environment:
+    TERM: "dumb"
+    ADB_INSTALL_TIMEOUT: 5000
+    GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
+
+general:
+  artifacts:
+    - "/home/ubuntu/android-base/app/build/reports"
+    - "/home/ubuntu/android-base/app/build/spoon"
 
 dependencies:
   pre:
+    - chmod +x app/set_animation_permissions.sh
     - echo y | android update sdk --no-ui --all --filter "tools,platform-tools,android-25,extra-google-m2repository"
     - echo y | android update sdk --no-ui --all --filter "build-tools-25.0.1"
     - echo y | android update sdk --no-ui --all --filter "extra-android-m2repository"
 
+  post:
+    # Create the android emulator
+    - echo n | android create avd -n test_android_17 -f -t android-17
```

Comment:
android 17 should be way faster than build in circleci-android22.
There is no difference between 17 and 22 in term of behaviour. 
New permission system was introduced in 23.